test/org/freedesktop/gstreamer/PropertyTypeTest.java

```diff
@@ -162,11 +161,27 @@ public void setEnum() {
         assertEquals("Red (brownian) noise", redNoise);
         audiotestsrc.setAsString("wave", redNoise);
         assertEquals(10, audiotestsrc.get("wave"));
+        
+        // invalid value
+        audiotestsrc.set("wave", -256);
+        assertEquals(0, audiotestsrc.get("wave"));
     }
     
     @Test(expected = IllegalArgumentException.class)
-    public void setEnumInvalid() {
-        audiotestsrc.set("wave", "FOO");
+    public void setEnumInvalidString() {
+        audiotestsrc.setAsString("wave", "FOO");
+    }
+    
+    @Test
+    public void setValueArrayFromString() {
+        Element convert = ElementFactory.make("audioconvert", null);
+        convert.setAsString("mix-matrix", "<<(float)0.25, (float)0.45>,<(float)0.65, (float)0.85>>");
+        String matrix = convert.getAsString("mix-matrix");
+        assertTrue(matrix.contains("0.2"));
+        assertTrue(matrix.contains("0.4"));
+        assertTrue(matrix.contains("0.6"));
+        assertTrue(matrix.contains("0.8"));
+        convert.dispose();
```

Comment:
The other Elements are creared in @Before and not disposed explicitly. We should harmonize this.

Reply:
I'm alright with the mix at the moment, as this element is only needed within that one test.  Explicit disposal is generally preferred - in many respects I wish the bindings didn't support implicit collection, but we are where we are!  Will keep at the moment, but relook later - tests need some love in general.